developer-hub/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+//!    it is easier to maintain and more reachable. Then, use excessive linking to back-link when
+//!    writing in a more high level.
+//!
+//! > A prime example of this, the details of the FRAME storage APIs should NOT be explained in a
+//! > high level tutorial. They should be explained in the rust-doc of the corresponding type or
+//! > macro.
+//!
+//! 2. 🧘 Less is More: For reasons mentioned [above](#crate::why-rust-docs), the more concise this
+//!    crate is, the better.
+//! 3. √ Don’t Repeat Yourself – DRY: A summary of the above two points. Authors should always
+//!    strive to avoid any duplicate information. Every concept should ideally be documented in
+//!    *ONE* place and one place only. This makes the task of maintaining topics significantly
+//!    easier.
+//!
+//! > A prime example of this, the list of CLI arguments of a particular binary should not be
+//! > documented in multiple places across this crate. It should be only be documented in the
+//! > corresponding crate (eg. `sc_cli`).
+//!
+//! > Moreover, this means that as a contributor, **it is your responsibility to have a grasp over
+//! > what topics are already covered in this crate, and how you can build on top of the information
+//! > that they already pose, rather than repeating yourself**.
+//!
+//! For more details about documenting guidelines, see:
+//! <https://github.com/paritytech/polkadot-sdk/master/docs/DOCUMENTATION_GUIDELINES.md>
+//!
+//! #### Example: Explaining `#[pallet::call]`
+//!
+//! Let's consider the seemingly simple example of explaining to someone dead-simple code of a FRAME
+//! call and see how we can use the above principles.
+//!
+//! ```
+//! #[frame::pallet]
+//! pub mod pallet(dev_mode) {
+//! #   use frame::prelude::*;
+//! #   #[pallet::config]
+//! #   pub trait Config: frame_system::Config {}
+//! #   #[pallet::pallet]
+//! #   pub struct Pallet<T>(_);
+//!     #[pallet::call]
+//!     impl<T: Config> Pallet<T> {
+//!         fn a_simple_call(origin: OriginFor<T>, data: u32) -> DispatchResult {
+//!             ensure!(data > 10, "SomeStaticString")
+//!             todo!();
+//!         }
+//!     }
+//! }
+//! ```
+//!
+//! * Before even getting started, what is with all of this `<T: Config>`? We link to
+//! [`reference_docs::trait_based_programming`].
+//! * First, the name. Why is this called `pallet::call`? This goes back to `enum Call`, which is
+//! explained in [`reference_docs::frame_composite_enums`]. Build on top of this!
+//! * Then, what is origin? Just an account id? [`reference_docs::origin_account_abstraction`].
+//! * Then, what is `DispatchResult`? why is this called *dispatch*? Probably something that can be
+//! explained in the documentation of [`frame::prelude::DispatchResult`].
+//! * Why is `"SomeStaticString"` a valid error? because of
+//!   [this](frame::prelude::DispatchError#impl-From<%26'static+str>-for-DispatchError).
```

Comment:
In general I would avoid having links that do not preview what you should expect to see if you follow it
```suggestion
//! * Why is `"SomeStaticString"` a valid error? Because there is an implicit conversion that you
//!   can see [here](frame::prelude::DispatchError#impl-From<%26'static+str>-for-DispatchError).
```

Reply:
Okay, but I edited your comment a bit.. I don't think "implicit" is correct here.